tests/test_boot.py

```diff
@@ -140,23 +138,73 @@ def _cleanup_boom_root_path(self, boot_path):
     def setUp(self):
         log.debug("Preparing %s", self._testMethodName)
         boom.set_boot_path("/boot")
+
+
+    def tearDown(self):
+        log.debug("Tearing down %s", self._testMethodName)
+        boom.set_boot_path(BOOT_ROOT_TEST)
+        self._clear_fstab()
+
+
+@unittest.skipIf(not have_root(), "requires root privileges")
+class BootTestsNonRoot(BootTestsBase):
+
+    def setUp(self):
+        self.volumes = ["root", "home", "var"]
+        self.thin_volumes = ["opt", "srv"]
+        self.boot_volumes = [
+                ("root", "/"),
+                ("home", "/home"),
+                ("var", "/var"),
+        ]
+        self._set_fstab()
         self._lvm = LvmLoopBacked(self.volumes, thin_volumes=self.thin_volumes)
-        snapset_name = "bootset0"
-        snapset_time = 1707923080
         for origin, mp in self.boot_volumes:
+            if origin == "root":
```

Comment:
Conceptually this is what we want to happen (excluding root from the snapshots created for the fake snapshot set), but I think it's possible to avoid the code duplication. E.g. we could have a flag set in the child class that triggers this behaviour, but keep the method definition in the base class (say `BootTestsBase.with_root=True` but `BootTestsNonRoot.with_root=False`).